impl/src/main/java/org/jboss/weld/bean/proxy/InterceptedProxyFactory.java

```diff
@@ -75,6 +78,28 @@ protected String getProxyNameSuffix() {
         return PROXY_SUFFIX + suffix;
     }
 
+    @Override
+    public void addInterfacesFromTypeClosure(Set<? extends Type> typeClosure, Class<?> proxiedBeanType) {
```

Comment:
Theoretically, this covers the case when user supplies `FooInterface` for `FooImpl` class and adds a class-level annotation and at the same time there is a `SuperFooInterace` with some default method.
The above (crazy) case needs to have a test added before we merge this.